### PR TITLE
Backend and Frontend Authentication Functional

### PR DESCRIPTION
**Pull request by Kaxuo**

### CODE REVIEW

File: FrontEnd/src/index.html [hunk 1]
```diff
@@ -1,13 +1,52 @@
...
+    />
+    <!-- Material Design Bootstrap -->
+    <link
+      href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.19.1/css/mdb.min.css"
+      rel="stylesheet"
+    />
+  </head>
+  <body class="body">
+    <app-root></app-root>
+    <!-- JQuery -->
```
Comment (jeremykairis) [inline FrontEnd/src/index.html]: You should avoid jQuery in Angular apps and use npm to add dependencies.
I checked mdbbootstrap and found their pricing annoying so it's better to learn [Angular Material](https://material.angular.io/) and [Bootstrap](https://getbootstrap.com/).

File: FrontEnd/src/app/components/profile-page/profile-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            this.CityEdit = false;
+          }
+          if (values.hobby) {
+            this.data.hobby = values.hobby;
+            this.hobbyEdit = false;
+          }
+          if (values.number) {
+            this.data.number = values.number;
+            this.numberEdit = false;
+          }
```
Comment (jeremykairis) [inline FrontEnd/src/app/components/profile-page/profile-page.component.ts]: Could you simplify this code being more generic ?

File: FrontEnd/src/app/components/profile-page/profile-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                Validators.required,
+              ]),
+            });
+            this.city = new FormGroup({
+              city: new FormControl(this.data.city, [Validators.required]),
+            });
+            this.hobby = new FormGroup({
+              hobby: new FormControl(this.data.hobby, [Validators.required]),
+            });
+          });
```
Comment (jeremykairis) [inline FrontEnd/src/app/components/profile-page/profile-page.component.ts]: Do you really need all these flags and separate FormGroups ?

File: FrontEnd/src/app/components/navbar/navbar.component.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (jeremykairis) [inline FrontEnd/src/app/components/navbar/navbar.component.scss]: Remove unused code.

File: BackEnd/Profile/AutoMapperProfile.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace BackEnd.Profile
+{
+    public class AutoMapperProfile : AutoMapper.Profile
+    {
+        public AutoMapperProfile()
+        {
+            CreateMap<User, UserModel>();
+            CreateMap<RegisterModel, User>();
+            CreateMap<UpdateModel, User>();
```
Comment (jeremykairis) [inline BackEnd/Profile/AutoMapperProfile.cs]: Try to simplify the user update process to reduce mappers.

File: BackEnd/Migrations/UserContextModelSnapshot.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+
+namespace BackEnd.Migrations
+{
+    [DbContext(typeof(UserContext))]
+    partial class UserContextModelSnapshot : ModelSnapshot
+    {
+        protected override void BuildModel(ModelBuilder modelBuilder)
+        {
+#pragma warning disable 612, 618
```
Comment (jeremykairis) [inline BackEnd/Migrations/UserContextModelSnapshot.cs]: Why do you use this #pragma ?

File: BackEnd/appsettings.json [hunk 7]
```diff
@@ -1,4 +1,10 @@
 {
+  "ConnectionStrings": {
+    "DefaultConnection": "server=localhost;port=3306;database=csharp;user=root;password=Da4EhBEQ5FiUmleVeFF1;CharSet=utf8"
+  },
+  "AppSettings": {
+    "Secret": "THIS IS USED TO SIGN AND VERIFY JWT TOKENS, REPLACE IT WITH YOUR OWN SECRET, IT CAN BE ANY STRING"
+  },
```
Comment (jeremykairis) [inline BackEnd/appsettings.json]: Sensitive data should be kept out of the source control.

**Review by jeremykairis — CHANGES_REQUESTED** [review]
Please, take the time to cleanup your code before submitting a PR. 
Rename the branch and the PR as it contains code from the backend. 
Use a proper .gitignore for the backend.
Remove extra white spaces and ensure clarity and consistency of variable names to be able to remove most comments.
Finally, try to keep the code as simple as possible.

**Comment by Kaxuo** [comment]
Everything should have been changed like you asked (except the name of the branch, I will try to be more accurate next time)

- Codes for the Profile Page has been totally changed, flags are added differently (and thus, not seen) , the "sendData" method doesn't have a "if" for each field anymore and is more generic. The FormGroup is more consice, no more FormGroup for each input field. ( commit : "Changes done to the frontend" )
Sorry, I did that at first because I couldn't find a solution for a "Single Field only Input" so I just coded it the hard way.

- Added some lines in the gitignore file to ignore what needs to be ignored in the subfolder "BackEnd"
(commit : "Remove ignored files")

- Sensitive Data are now stored safely and can no longer be seen in the BackEnd , appsetting.json
(commit : "ChangesRequested Done")

- Deleted the UpdateModel because it wasn't needed anymore. 
(commit : "CleanUp")

- #Pragma file is automatically generated when migrating and updating the database.

I hope I did everything correctly. Sorry if I took some time to do it. I got stuck trying to find a way to hide sensitive data when the solution was so simple.

Many Thanks

File: BackEnd/Controllers/UserController.cs [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+        }
+        [HttpGet]
+        public ActionResult<IEnumerable<User>> GetAll()
+        {
+            var users = _repository.GetAllUsers();
+            var model = _mapper.Map<IList<UserModel>>(users);
+            return Ok(model);
+        }
+
+
```
Comment (jeremykairis) [inline BackEnd/Controllers/UserController.cs]: Be consistent using white spaces.

File: BackEnd/Data/MySQLMethods.cs [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        public IEnumerable<User> GetAllUsers()
+        {
+            return _context.Users;
+        }
+
+        public User GetUserById(int id)
+        {
+            return _context.Users.Find(id);
+        }
+
```
Comment (jeremykairis) [inline BackEnd/Data/MySQLMethods.cs:71]: Take care of white spaces to make clean code.

File: BackEnd/Data/UserContext.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        public UserContext(DbContextOptions<UserContext> opt) : base(opt)
+        {
+        }
+
+        public UserContext()
+        {
+        }
+
+        // protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (jeremykairis) [inline BackEnd/Data/UserContext.cs]: Remove the old code.

File: FrontEnd/src/index.html [hunk 11]
```diff
@@ -1,13 +1,32 @@
...
-<body>
-  <app-root></app-root>
-</body>
+  <head>
+    <meta charset="utf-8" />
+    <title>FrontEnd</title>
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
+    <link
```
Comment (jeremykairis) [inline FrontEnd/src/index.html]: Can you use npm to add these links ?

**Review by jeremykairis — CHANGES_REQUESTED** [review]
It gets better and better, continue to improve the .gitignore and cleanup white spaces to get it merged.

**Comment by Kaxuo** [comment]
- White spaces have been cleanup a little bit better ( hopefully ) and made more consistent
- Sorry about unused codes, most of the times, they're here for me as a reminder that either I need to check up on that later or that it "exists" so that I could use it again somewhere else.
- Links has been removed in the index.html, they're not needed. I probably added them when trying to make MDB work when a simple npm install would have sufficed
- I probably made a mistake, I did a git push -f origin and it closed the pull request. It says that It has a different history now, I can't find a way to re open it. I messed around and in the end , I just went ahead and forced push to the master.
I will be more careful next time for the next pull request. Sorry for the inconvenience, Thanks

**Pull request closed without merging**